website/docs/language/settings/backends/gcs.mdx

```diff
@@ -73,6 +73,28 @@ the path of the service account key. Terraform will use that key for authenticat
+
+### Customer-supplied encryption keys
+
+To get started, follow this guide: [Use customer-supplied encryption keys](https://cloud.google.com/storage/docs/encryption/using-customer-supplied-keys)
+
+If you want to remove customer-supplied keys from your backend configuration, or change to a different customer-supplied key, Terraform cannot perform a state migration automatically and manual intervention is needed instead. This is because customer-supplied encryption keys are not stored by Google, so they must be supplied in any requests sent to the Cloud Storage API (see [Customer-supplied Encryption Keys](https://cloud.google.com/storage/docs/encryption/customer-supplied-keys)). At the time of state migration, the old key's details are removed from the backend configuration and Terraform cannot use the key during the migration process.
+
+~> **Important:** To migrate your state away from using customer-supplied encryption keys, or change the key used by your backend, you will need to perform a [rewrite (gsutil CLI)](https://cloud.google.com/storage/docs/gsutil/commands/rewrite) or [cp (gcloud CLI)](https://cloud.google.com/sdk/gcloud/reference/storage/cp#--decryption-keys) operation to remove use of the old customer-supplied encryption key on your state file. Once the encryption is removed you can successfully run `terraform init -migrate-state` with your new backend configuration.
+
+### Customer-managed encryption keys (Cloud KMS)
+
+To get started, follow this guide: [Use customer-managed encryption keys](https://cloud.google.com/storage/docs/encryption/using-customer-managed-keys)
+
+If you want to remove customer-managed keys from your backend configuration, or change to a different customer-managed key, Terraform _can_ manage a state migration without manual intervention. This is because customer-managed encryption keys are stored in GCP and are accessible during the state migration process. However, these changes do not fully come into effect until the first write operation is performed on the state file after state migration occurs. In the first write operation after state migration the file will be decrypted with the old key and then written with the new encryption method. This is equivalent to the [rewrite](https://cloud.google.com/storage/docs/gsutil/commands/rewrite) operation described in the customer-supplied encryption keys section above. Because of the importance of the first write to state after state migration, you should not delete old KMS keys until any state file(s) encrypted with that key have been updated.
```

Comment:
```suggestion
If you want to remove customer-managed keys from your backend configuration or change to a different customer-managed key, Terraform _can_ manage a state migration without manual intervention. This ability is because GCP stores customer-managed encryption keys and are accessible during the state migration process. However, these changes do not fully come into effect until the first write operation occurs on the state file after state migration occurs. In the first write operation after state migration, the file decrypts with the old key and then writes with the new encryption method. This method is equivalent to the [rewrite](https://cloud.google.com/storage/docs/gsutil/commands/rewrite) operation described in the customer-supplied encryption keys section. Because of the importance of the first write to state after state migration, you should not delete old KMS keys until any state file(s) encrypted with that key update.
```

Suggest adding "ability" after "This" to avoid a "This" with no subject. Suggest changing "are stored in GCP" to "GCP stores" to avoid a passive voice style issue. Suggest changing "will be decrypted" and "is performed" to "decrypts" and "occurs" to avoid future tense and passive voice style issues. Suggest changing "have been updated" to "update" to avoid a passive voice style issue.